autoload/neomake.vim

```diff
@@ -111,8 +111,13 @@ function! s:MakeJob(make_id, maker) abort
 
     if has_key(a:maker, 'cwd')
         let old_wd = getcwd()
+        if has('nvim')
+            let chdir = haslocaldir() ? 'lcd' : haslocaldir(-1, 0) ? 'tcd' : 'cd'
+        else
+            let chdir = haslocaldir() ? 'lcd' : 'cd'
+        endif
         let cwd = expand(a:maker.cwd, 1)
-        exe 'cd' fnameescape(cwd)
+        exe chdir . ' ' . fnameescape(cwd)
```

Comment:
FWIW, I'd prefer to use `cd` for the var name and then just `exe cd fnameescape(cwd)`.